cmd/lakectl/cmd/config/config.go

```diff
@@ -42,20 +46,36 @@ type configuration struct {
 // Config is the currently-loaded configuration.  Its error state supports being able to run
 // `lakectl config' without a valid configuration.
 type Config struct {
-	configuration
-	err error
+	Values configuration
+	err    error
 }
 
 // ReadConfig loads according to the current viper configuration into a Config, which will
 // have non-nil Err() if loading fails.
 func ReadConfig() (c *Config) {
 	c = &Config{}
+
+	// Inform viper of all expected fields.  Otherwise it fails to deserialize from the
+	// environment.
+	keys := config.GetStructKeys(reflect.TypeOf(c.Values), "mapstructure", "squash")
+	for _, key := range keys {
+		viper.SetDefault(key, nil)
+	}
+
 	setDefaults()
 	c.err = viper.ReadInConfig()
+	logger := logging.Default().WithField("file", viper.ConfigFileUsed())
+
+	if c.err == nil {
+		logger.Info("loaded configuration from file")
+	} else if _, ok := c.err.(viper.ConfigFileNotFoundError); !ok {
+		logger.WithError(c.err).Fatal("failed to read config file")
+	}
+
+	c.err = viper.UnmarshalExact(&c.Values)
```

Comment:
looks like this 'err' can shade the actual error done in the loading process.
think this one should move to the pre-command stage.